### PR TITLE
Unique mixes

### LINKED ISSUE #79: Duplicated mixes in case of declared entity
При миксировании сущности в JSX, происходит дублирование микса.
Пример:

payment-scenario.react.js
```js
import React from 'react';
import { decl } from 'bem-react-core';

export default decl({
    block: 'payment-scenario'
});
```

Вызов
```js
<PaymentScenario mix={{ block: 'island', elem: 'section' }} />
```

=>

Получаемый HTML
```html
<div class="payment-scenario island__section island__section"></div>
```

### PR DESCRIPTION
**Pull request by awinogradov**
Close #79

### CODE REVIEW

File: lib/ClassNameBuilder.js [hunk 1]
```diff
@@ -8,27 +8,24 @@ export default class ClassNameBuilder {
 
     stringify(block, elem, mods, mixes, cls) {
         return this.b_(block, elem, mods) +
-            (mixes? ' ' + mixes.map(mix => mix?
+            (mixes && mixes.length? ' ' + mixes.map(mix => mix?
                 this.b_(mix.block || block, mix.elem, mix.mods) : '').join(' ') :
                 '') + (cls? ' ' + cls : '');
     }
 
-    joinMixes(mix1, mix2) {
```
Comment (awinogradov) [inline lib/ClassNameBuilder.js]: @dfilatov do you remember why you called this function as `joinMixes`? And why this function used 2 arguments?

Comment (dfilatov) [inline lib/ClassNameBuilder.js]: History says that you named it `joinMixes`, originally it was named as `buildMixes` (https://github.com/bem/bem-react-core/commit/3b9da98b7daae5008087245428ac2baa78a49f7a#diff-168726dbe96b3ce427e7fedce31bb0bcL17)

Comment (awinogradov) [inline lib/ClassNameBuilder.js]: Ok ;) But why 2 arguments?

Comment (dfilatov) [inline lib/ClassNameBuilder.js]: Could you reformulate question? I don't understand it. How many arguments do you expect?

Comment (awinogradov) [inline lib/ClassNameBuilder.js]: We changed 2 arguments to 1. And I trying to find bad things in this changes. All tests passed, but maybe, you can remember something about this function.

Comment (awinogradov) [inline lib/ClassNameBuilder.js]: Thanks! It's better now ;)

File: __tests__/fixtures1.blocks/BlockWithDeclaredMix/BlockWithDeclaredMix.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { decl } from 'bem-react-core';
+
+export default decl({
+    block : 'BlockWithDeclaredMix',
+    mix() {
```
Comment (awinogradov) [inline __tests__/fixtures1.blocks/BlockWithDeclaredMix/BlockWithDeclaredMix.js]: I don't understand difference between `mix` and `addMix`. Which one I need to use in declaration?

Comment (tadatuta) [inline __tests__/fixtures1.blocks/BlockWithDeclaredMix/BlockWithDeclaredMix.js]: `mix(val)` is like `mix = val` and `addMix(val)` is like `mix.push(val)` so you need to use `addMix()` if you want to add new mix and don't want to override old value

Comment (awinogradov) [inline __tests__/fixtures1.blocks/BlockWithDeclaredMix/BlockWithDeclaredMix.js]: @tadatuta what about `...this.__base(...arguments)`? I understand what you mean, but this helper isn't connected to this issue. If want make similar helpers we need to have: `addMix`, `addAttrs`, `addCls`, `addMods`, `addStyle` and etc.

Comment (tadatuta) [inline __tests__/fixtures1.blocks/BlockWithDeclaredMix/BlockWithDeclaredMix.js]: I guess we really need the same collection of helpers as we have in `bem-xjst`

Comment (awinogradov) [inline __tests__/fixtures1.blocks/BlockWithDeclaredMix/BlockWithDeclaredMix.js]: https://github.com/bem/bem-react-core/issues/83

**Pull request merged**